smart-contracts/contracts/mixins/MixinRefunds.sol

```diff
@@ -190,18 +193,23 @@ contract MixinRefunds is
     Key storage key = _getKeyFor(_owner);
     // Math: safeSub is not required since `hasValidKey` confirms timeRemaining is positive
     uint timeRemaining = key.expirationTimestamp - block.timestamp;
-    if(timeRemaining >= expirationDuration) {
+    if(timeRemaining + freeTrialLength >= expirationDuration) {
       refund = keyPrice;
     } else {
       // Math: using safeMul in case keyPrice or timeRemaining is very large
       refund = keyPrice.mul(timeRemaining) / expirationDuration;
     }
-    uint penalty = keyPrice.mul(refundPenaltyNumerator) / refundPenaltyDenominator;
-    if (refund > penalty) {
-      // Math: safeSub is not required since the if confirms this won't underflow
-      refund -= penalty;
-    } else {
-      refund = 0;
+
+    // Apply the penalty if this is not a free trial
+    if(freeTrialLength == 0 || timeRemaining + freeTrialLength < expirationDuration)
```

Comment:
simplify with `timeRemaining + freeTrialLength < expirationDuration` only?

Reply:
That changes a rare scenario a bit.
If I buy a key from a lock that does not offer trials.  Then I buy a second key (either normally or via a marketplace).  Now my timeRemaining is > the default expirationDuration... however the lock owner would still want the base penalty to apply on refund. (i suspect..)